### PR TITLE
fix: joi validations

### LINKED ISSUE #76: Add missing Joi validations
### Tasks

- [ ] Add/enable Joi validation for `exportCommitments` options
- [ ] Add/enable Joi validation for `importAndSaveCommitments` options
- [ ] Enhance `makeTransferOptions.nightfallRecipientAddress` - add a "custom validation"

Validations live in libs/user/validations.ts file. Enable validations in user/user.ts file.

For validating nightfall addresses with a custom validation function just follow the `isChecksum` implementation.

### LINKED ISSUE #90: Review Joi validations
Although it is not a bug, current Joi validations "fail silently".

This means that validation errors are not being reported back to the SDK. That's because of a misinterpretation about how the library works. See https://joi.dev/api/?v=17.6.0#general-usage. The idea is to capture the results of the `validate` method then "do something about it".

### Goal

Capture validation errors as per the Joi docs, and agree/make a proposal about what to do with these.

### PR DESCRIPTION
**Pull request by imagobea**
## What does this implement/fix? Explain your changes.

- Fixes Joi output management by handling errors/re-using formatted parameters

- Refactors existing Joi validations

- Introduces `whichTokenStandard` for interface id detection, ie you no longer need to provide `tokenErcStandard` (this logic was sitting on txDeposit script). Has coverage.

- Introduces `prepareTokenValueTokenId` for interface id detection, setting tx defaults, setting token and converting values. This removes some duplicate code in transactions. Has coverage.

- Introduces new env vars :warning: 

- Introduces `mint721` script for minting token Ids before transactiong 721s (this logic was sitting on txDeposit script). Package.json updated accordingly.

- Cleans up txDeposit script

- Updates old instances of "units" file (util was renamed as `stringValueToWei` some time ago)

- Updates `checkStatus` in favour of `isClientAlive`, `isWeb3WsAlive`

### Summary of breaking changes for Nightfall users

None for now since I kept both `tokenErcStandard` att in transactions, and `checkStatus` method for compatibility, but would be good to think how do we manage these sort of changes and when do we deprecate them.

## Does this close any currently open issues?

Closes #90

## What commands can I run to test the change? 

You can run tests/example scripts as per the commands on the package.json

## Any other comments?

n/a

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -84,6 +84,7 @@
     "eg:goerli:export-commitments": "ts-node -r dotenv/config examples/scripts/exportCommitments.ts dotenv_config_path=./examples/scripts/.env.goerli",
     "eg:goerli:import-commitments": "ts-node -r dotenv/config examples/scripts/importCommitments.ts dotenv_config_path=./examples/scripts/.env.goerli",
     "eg:web-app": "npm run build && cd examples/webApp/ && npm run serve",
-    "eg:react-web-app-start": "npm run build && cd examples/web-app/ && npm start"
+    "eg:react-web-app-start": "npm run build && cd examples/web-app/ && npm start",
+    "utils:ganache:mint-721": "ts-node -r dotenv/config examples/scripts/utils/mint721.ts dotenv_config_path=./examples/scripts/.env.ganache"
```
Comment (velkoskis) [inline package.json:88]: love the mint script +1

File: libs/tokens/constants.ts [hunk 2]
```diff
@@ -11,3 +7,5 @@ export const ERC20 = "ERC20";
 export const ERC721 = "ERC721";
 export const ERC1155 = "ERC1155";
 export const ERC165 = "ERC165";
+
+export const ERC721_INTERFACE_ID = "0x80ac58cd";
```
Comment (velkoskis) [inline libs/tokens/constants.ts:11]: nice, good call adding this

**Review by velkoskis — APPROVED** [review]
LGTM! Will make my job easier for transfer and withdrawal refactor :)

**Comment by imagobea** [comment]
> LGTM! Will make my job easier for transfer and withdrawal refactor :)

You're far too nice with me! :nerd_face:

**Pull request merged**